### PR TITLE
Single video page

### PR DESCRIPTION
**Pull request by bhtibrewal**
In this pr:
- added toast to all functions
- added 404page and user-profile page
- added search filter

Pages to avoid:
- all CSS files
- index.js files

### CODE REVIEW

File: src/services/playlists/removeFromPlaylist.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
 import axios from "axios"
 
-export const removeFromPlaylist = async ({ videoId, playlistId, videoStateDispatch }) => {
+export const removeFromPlaylist = async ({ videoId, playlistId, videoStateDispatch, showToast }) => {
     try {
-        const {data: {playlist}} = await axios.delete(`/api/user/playlists/${playlistId}/${videoId}`);
-        videoStateDispatch({ type: "HANDLE_PLAYLIST", payload: playlist });
+        const res = await axios.delete(`/api/user/playlists/${playlistId}/${videoId}`);
+        if (res.status === 200) {
+            videoStateDispatch({ type: "HANDLE_PLAYLIST", payload: res.data.playlist });
```
Comment (Naman-Saxena1) [inline src/services/playlists/removeFromPlaylist.js:7]: type : HANDLE_PLAYLIST can be improved to UPDATE_PLAYLIST or something more descriptive.

**Review by Naman-Saxena1 — APPROVED** [review]
Code and UI looks great with best practices being followed 👍 
Overall good to merge 
Later, can work on responsiveness, share button is overflowing for smaller screen
Capture
s

File: src/pages/user_profile_page/user_profile.css [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  text-align: center;
+}
+.user-list-sec{
+  display:flex;
+  flex-direction: column;
+  gap: 2rem;
+}
+.sec {
+  background: var(--light-bg);
+  padding: 2rem 2rem;
```
Comment (Sahil25061999) [inline src/pages/user_profile_page/user_profile.css:16]: you could just use padding: 2rem here.

**Review by Sahil25061999 — APPROVED** [review]
just suggested a minor change, the search functionality is working great.
The code looks good, nice use of comments .
UI suggestion: for the homepage you can increase the margin between the cards and the categories text.

File: src/pages/user_profile_page/user_profile.css (hunk 2 above)
Comment (bhtibrewal) [reply src/pages/user_profile_page/user_profile.css:16]: done fixed

**Pull request merged**